app/controllers/items_controller.rb

```diff
@@ -693,16 +692,16 @@ def save_and_reindex
   # Permissions
 
   # check that the user can carry out this item modification
-  def forbid_modify
-    return true if current_user.is_admin || @object.can_manage_content?(current_user.roles(@apo))
-    render :status => :forbidden, :text => 'forbidden'
-    false
+  def authorize_manage_obj!
+    authorize! :manage_item, @object
   end
```

Comment:
The original design broke out the parts of the object — desc MD, sys MD, contents, rights, WF — in order to give maximum flexibility in crafting roles with various restrictions.  As you can see from the resulting matrix (https://consul.stanford.edu/display/chimera/Repository+Roles+and+Permissions), the critical differences that resulted and that haven’t changed were meta roles (governing items) vs APO roles (governed items) and the distinction between manage (write) and view (read).  But otherwise it’s all-or-nothing access to the items you see, making the difference between manage_item vs manage_content moot since the first includes the second for the roles we’ve implemented.

We still think there might be a place someday for the -metadata role as the audience expands to folks from the metadata unit, a much lower chance that we’d find a “you can manage the DOR object but not its content files”  ever useful. So it’s probably not worth making such distinctions in the code until we actually need them. I can collapse the table on the consul page to better indicate the parts of the object covered without implying we support that granularity of access if that helps.

Lynn


> On Feb 26, 2016, at 11:38 AM, Chris Beer <notifications@github.com> wrote:
> 
> In app/controllers/items_controller.rb:
> 
> > @@ -693,16 +692,16 @@ def save_and_reindex
> >    # Permissions
> >  
> >    # check that the user can carry out this item modification
> > -  def forbid_modify
> > -    return true if current_user.is_admin || @object.can_manage_content?(current_user.roles(@apo))
> > -    render :status => :forbidden, :text => 'forbidden'
> > -    false
> > +  def authorize_manage_obj!
> > +    authorize! :manage_item, @object
> >    end
> 
> Thanks, manage_item vs manage_content was a mistake on my part. I'm interested in understanding more about when is_manager implies manage_content and when it does not. Maybe there's something deeper going on. @LynnMcRae ?
> 
> —
> Reply to this email directly or view it on GitHub.
>